src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -833,7 +838,22 @@ public ImmutableArray<T> RemoveRange(ImmutableArray<T> items)
         [Pure]
         public ImmutableArray<T> RemoveRange(ImmutableArray<T> items, IEqualityComparer<T> equalityComparer)
         {
-            return this.RemoveRange(items.array, equalityComparer);
+            var self = this;
+            Requires.NotNull(items.array, "items");
+
+            if (items.IsEmpty)
+            {
+                self.ThrowNullRefIfNotInitialized();
+                return self;
+            }
+            else if (items.Length == 1)
+            {
+                return self.Remove(items[0], equalityComparer);
```

Comment:
@sharwell I was thinking about [Array.CreateInstance](https://msdn.microsoft.com/en-us/library/x836773a(v=vs.110).aspx) but as @stephentoub points out that doesn't let you then cast the resulting `System.Array` to `T[]` so it's not relevant to this discussion after all.